examples/generate.sh

```diff
@@ -114,16 +114,20 @@ echo "Generated ${MACHINEDEPLOYMENT_GENERATED_FILE}"
 # Get Cert-manager provider components file
 curl -L -o "${COMPONENTS_CERT_MANAGER_GENERATED_FILE}" https://github.com/jetstack/cert-manager/releases/download/v0.13.0/cert-manager.yaml
 
+# Get enhanced envsubst version to evaluate expressions like ${VAR:=default}
+curl -L -o envsubst-go https://github.com/a8m/envsubst/releases/download/v1.2.0/envsubst-`uname -s`-`uname -m`
+chmod +x envsubst-go
+
 # Generate Cluster API provider components file.
-"$KUSTOMIZE" build "github.com/kubernetes-sigs/cluster-api/config/?ref=master" > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
+"$KUSTOMIZE" build "github.com/kubernetes-sigs/cluster-api/config/?ref=master" | ./envsubst-go > "${COMPONENTS_CLUSTER_API_GENERATED_FILE}"
```

Comment:
most variables in the manifests are not defined when we call envsubst. You could define a set of default variable used just to populate the example manifests.

Reply:
Yes, it should be like that :)
By the way, do you think that replacing "vanilla" envsubst entirely with the "enhanced" version could be a thing there?
With the PR we will use two different envsubst when one of them just supports both use cases.

Reply:
I didn't quite understood you probably, added the following just in case 
https://github.com/metal3-io/cluster-api-provider-metal3/pull/126/commits/6bbfb1b8d5d90b43fff5df66e8ea13c99ac1d328#diff-2b085cd4a651deec0d4f1b1850814f02R43-R44

Reply:
I didn't see that they are now with a default within the manifests, sorry, my bad.

Reply:
yes, I will remove default values from the generate.sh then